### PR TITLE
TST: assert rewriting in tests

### PR DESCRIPTION
**Pull request by enricofer**
ref FOSS4G2022 codesprint

### CODE REVIEW

File: tests/test_mapping.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-        self.assertEqual(m['type'], 'Point')
-        self.assertEqual(m['coordinates'], (0.0, 0.0))
-        
+        assert m['type'] == 'Point'
+        assert m['coordinates'] == (0.0, 0.0)
+
     def test_empty_polygon(self):
         """Empty polygons will round trip without error"""
-        self.assertIsNotNone(mapping(Polygon()))
+        assert mapping(Polygon()) != None
```
Comment (mwtoews) [inline tests/test_mapping.py:13]:
```suggestion
        assert mapping(Polygon()) is not None
```

File: tests/test_polygonize.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def test_polygonize(self):
             Point(0, 0),
         ]
         result = list(polygonize(lines))
-        self.assertTrue(all([isinstance(x, Polygon) for x in result]))
+        assert all([isinstance(x, Polygon) for x in result])
```
Comment (mwtoews) [inline tests/test_polygonize.py:19]:
```suggestion
        assert all(isinstance(x, Polygon) for x in result)
```

File: tests/test_polygonize.py [hunk 3]
```diff
@@ -31,12 +31,10 @@ def test_polygonize_full(self):
         ]
 
         result2, dangles, cuts, invalids = polygonize_full(lines2)
-        self.assertEqual(len(result2.geoms), 2)
-        self.assertTrue(all([isinstance(x, Polygon) for x in result2.geoms]))
-        self.assertEqual(list(dangles.geoms), [])
-        self.assertTrue(all([isinstance(x, LineString) for x in cuts.geoms]))
+        assert len(result2.geoms) == 2
+        assert all([isinstance(x, Polygon) for x in result2.geoms])
```
Comment (mwtoews) [inline tests/test_polygonize.py:35]:
```suggestion
        assert all(isinstance(x, Polygon) for x in result2.geoms)
```

File: tests/test_polygonize.py [hunk 4]
```diff
@@ -31,12 +31,10 @@ def test_polygonize_full(self):
...
+        assert list(dangles.geoms) == []
+        assert all([isinstance(x, LineString) for x in cuts.geoms])
```
Comment (mwtoews) [inline tests/test_polygonize.py:37]:
```suggestion
        assert all(isinstance(x, LineString) for x in cuts.geoms)
```

File: tests/test_polylabel.py [hunk 5]
```diff
@@ -35,17 +37,17 @@ def test_cell_sorting(self):
...
-        self.assertFalse(cell1 == cell2)
-        self.assertNotEqual(cell1, cell2)
-        self.assertFalse(cell1 != cell1)
-        self.assertGreater(cell2, cell1)
-        self.assertFalse(cell1 > cell2)
-        self.assertGreaterEqual(cell2, cell1)
-        self.assertFalse(cell1 >= cell2)
+        assert cell1 < cell2
+        assert cell1 <= cell2
+        assert (cell2 <= cell1) == False
```
Comment (mwtoews) [inline tests/test_polylabel.py:42]:
```suggestion
        assert (cell2 <= cell1) is False
```

File: tests/test_polylabel.py [hunk 6]
```diff
@@ -35,17 +37,17 @@ def test_cell_sorting(self):
...
+        assert cell1 == cell1
+        assert (cell1 == cell2) == False
```
Comment (mwtoews) [inline tests/test_polylabel.py:44]:
```suggestion
        assert (cell1 == cell2) is False
```

File: tests/test_polylabel.py [hunk 7]
```diff
@@ -35,17 +37,17 @@ def test_cell_sorting(self):
...
+        assert cell1 != cell2
+        assert (cell1 != cell1) == False
```
Comment (mwtoews) [inline tests/test_polylabel.py:46]:
```suggestion
        assert (cell1 != cell1) is False
```

File: tests/test_polylabel.py [hunk 8]
```diff
@@ -35,17 +37,17 @@ def test_cell_sorting(self):
...
+        assert cell2 > cell1
+        assert (cell1 > cell2) == False
```
Comment (mwtoews) [inline tests/test_polylabel.py:48]:
```suggestion
        assert (cell1 > cell2) is False
```

File: tests/test_polylabel.py [hunk 9]
```diff
@@ -35,17 +37,17 @@ def test_cell_sorting(self):
...
+        assert cell2 >= cell1
+        assert (cell1 >= cell2) == False
```
Comment (mwtoews) [inline tests/test_polylabel.py:50]:
```suggestion
        assert (cell1 >= cell2) is False
```

File: tests/test_styles.py [hunk 10]
```diff
@@ -5,11 +5,11 @@
...
+        assert CAP_STYLE.flat == 2
+        assert CAP_STYLE.square == 3
 
     def test_join(self):
-        self.assertEqual(JOIN_STYLE.round, 1)
-        self.assertEqual(JOIN_STYLE.mitre, 2)
-        self.assertEqual(JOIN_STYLE.bevel, 3)
+        assert JOIN_STYLE.round == 1
+        assert JOIN_STYLE.mitre == 2
+        assert JOIN_STYLE.bevel == 3
```
Comment (mwtoews) [inline tests/test_styles.py:15]: these changes will conflict with #1508, so changes to this file can be unstaged

File: tests/test_nearest.py [hunk 11]
```diff
@@ -1,15 +1,16 @@
...
-        self.assertAlmostEqual(first.x, 1.0, 7)
-        self.assertAlmostEqual(second.x, 2.0, 7)
-        self.assertAlmostEqual(first.y, 0.0, 7)
-        self.assertAlmostEqual(second.y, 0.0, 7)
+            Point(0, 0).buffer(1.0), Point(3, 0).buffer(1.0),
+        )
+        assert first.x == pytest.approx(1.0, 7)
+        assert second.x == pytest.approx(2.0, 7)
+        assert first.y == pytest.approx(0.0, 7)
+        assert second.y == pytest.approx(0.0, 7)
```
Comment (mwtoews) [inline tests/test_nearest.py:16]: the second parameter for `pytest.approx` is not the same for `assertAlmostEqual`. Try removing it:
```suggestion
        assert first.x == pytest.approx(1.0)
        assert second.x == pytest.approx(2.0)
        assert first.y == pytest.approx(0.0)
        assert second.y == pytest.approx(0.0)
```
if that is causing issues, then I believe an equivalent is `pytest.approx(1.0, abs=1e-7)`

**Review by mwtoews — COMMENTED** [review]
Thanks for the nice rewrites! A few suggested changes follows...

**Review by mwtoews — APPROVED** [review]
Thanks again. A next step would be to re-write the unittest class approach to use simple pytest constructs. This can happen later.

**Review by jorisvandenbossche — APPROVED** [review]
Thanks a lot @enricofer !

**Pull request merged**